middlewares/replacePath.go

```diff
@@ -0,0 +1,26 @@
+package middlewares
+
+import (
+	"net/http"
+)
+
+// ReplacePath is a middleware used to replace the path of a URL request
+type ReplacePath struct {
+	Handler http.Handler
+	Path    string
+}
+
+// ReplacedPathHeader is the default header to set the old path to
+const ReplacedPathHeader = "X-Replaced-Path"
+
+func (s *ReplacePath) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	r.Header.Add(ReplacedPathHeader, r.URL.Path)
+	r.URL.Path = s.Path
+	r.RequestURI = r.URL.RequestURI()
+	s.Handler.ServeHTTP(w, r)
+}
+
+// SetHandler sets handler
+func (s *ReplacePath) SetHandler(Handler http.Handler) {
```

Comment:
I was wondering the same thing for the `AddPrefix` handler. I thought it might be part of an interface so I copied it anyways.

I'll remove it since you don't think it's needed either.